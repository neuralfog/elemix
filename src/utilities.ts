export const ref = <Value>(
    value: Value | undefined = undefined,
): { value?: Value } => {
    return {
        value,
    };
};
