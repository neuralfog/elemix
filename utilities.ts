export { mergeClasses } from '@neuralfog/elemix-renderer/utils';
export * from './src/utilities';
