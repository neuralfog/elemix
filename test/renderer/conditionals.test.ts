import { expect, test, describe, beforeEach } from 'vitest';
import { html } from '@neuralfog/elemix-renderer';
import { Present } from '@neuralfog/elemix-testing';
import { render } from '../../utilities';
import { HTML } from '@neuralfog/elemix-testing/snapshots';
import {
    State,
    type ConditionalRendering,
} from '../fixtures/renderer/ConditionalRendering';

import '../fixtures/renderer/ConditionalRendering.ts';

describe('Render Conditionals', () => {
    beforeEach(() => {
        document.body.innerHTML = '';
    });

    test('single', async () => {
        // Initial State
        const presenter = new Present().screen(html`<conditional-rendering />`);
        const component = presenter.getComponent<ConditionalRendering>(
            'conditional-rendering',
        );
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();

        component.state.condition = false;
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();
    });

    test('start', async () => {
        const presenter = new Present().screen(html`<conditional-rendering />`);
        const component = presenter.getComponent<ConditionalRendering>(
            'conditional-rendering',
        );

        component.state.condition = true;
        component.state.case = State.START;
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();

        component.state.condition = false;
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();
    });

    test('end', async () => {
        const presenter = new Present().screen(html`<conditional-rendering />`);
        const component = presenter.getComponent<ConditionalRendering>(
            'conditional-rendering',
        );

        component.state.condition = true;
        component.state.case = State.END;
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();

        component.state.condition = false;
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();
    });

    test('middle', async () => {
        const presenter = new Present().screen(html`<conditional-rendering />`);
        const component = presenter.getComponent<ConditionalRendering>(
            'conditional-rendering',
        );

        component.state.condition = true;
        component.state.case = State.MIDDLE;
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();

        component.state.condition = false;
        await render();
        expect(HTML(presenter.root())).toMatchSnapshot();
    });
});
