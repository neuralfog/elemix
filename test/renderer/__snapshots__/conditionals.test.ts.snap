// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Render Conditionals > end 1`] = `
"
<conditional-rendering><!-- #shadowroot -->
  <p>hello there</p>
  <p>true</p>
</conditional-rendering>
"
`;

exports[`Render Conditionals > end 2`] = `
"
<conditional-rendering><!-- #shadowroot -->
  <p>hello there</p>
  <p>false</p>
</conditional-rendering>
"
`;

exports[`Render Conditionals > middle 1`] = `
"
<conditional-rendering><!-- #shadowroot -->
  <p>hello there</p>
  <p>true</p>
  <p>hello there</p>
</conditional-rendering>
"
`;

exports[`Render Conditionals > middle 2`] = `
"
<conditional-rendering><!-- #shadowroot -->
  <p>hello there</p>
  <p>false</p>
  <p>hello there</p>
</conditional-rendering>
"
`;

exports[`Render Conditionals > single 1`] = `
"
<conditional-rendering><!-- #shadowroot -->
  <p>true</p>
</conditional-rendering>
"
`;

exports[`Render Conditionals > single 2`] = `
"
<conditional-rendering><!-- #shadowroot -->
  <p>false</p>
</conditional-rendering>
"
`;

exports[`Render Conditionals > start 1`] = `
"
<conditional-rendering><!-- #shadowroot -->
  <p>true</p>
  <p>hello there</p>
</conditional-rendering>
"
`;

exports[`Render Conditionals > start 2`] = `
"
<conditional-rendering><!-- #shadowroot -->
  <p>false</p>
  <p>hello there</p>
</conditional-rendering>
"
`;
